src/main/kotlin/no/nb/mlt/wls/application/hostapi/item/ApiCreateItemPayload.kt

```diff
@@ -0,0 +1,130 @@
+      "itemCategory": "PAPER",
+      "preferredEnvironment": "NONE",
+      "packaging": "NONE",
+      "callbackUrl": "https://callback-wls.no/item"
+    }
+    """
+)
+data class ApiCreateItemPayload(
+    @Schema(
+        description = """The item ID from the host system, usually a barcode or an equivalent ID.""",
+        example = "mlt-12345"
+    )
+    val hostId: String,
+    @Schema(
+        description = """Name of the host system which the item originates from.
+                Host system is usually the catalogue that the item is registered in.""",
+        examples = [ "AXIELL", "ALMA", "ASTA", "BIBLIOFIL" ]
+    )
+    val hostName: HostName,
+    @Schema(
+        description = """Description of the item for easy identification in the warehouse system.
+                Usually an item title/name, e.g. book title, film name, etc. or contents description.""",
+        examples = ["Tyven, tyven skal du hete", "Avisa Hemnes", "Kill Buljo"]
+    )
+    val description: String,
+    @Schema(
+        description = """Item's category, same category indicates that the items can be stored together without any preservation issues.
+                For example: books, magazines, newspapers, etc. are of type PAPER, and can be stored together without damaging each other.""",
+        examples = ["PAPER", "DISC", "FILM", "PHOTO", "EQUIPMENT", "BULK_ITEMS", "MAGNETIC_TAPE"]
+    )
+    val itemCategory: ItemCategory,
+    @Schema(
+        description = """What kind of environment the item should be stored in.
+                "NONE" is for normal storage for the item category, "FRYS" is for frozen storage, etc.
+                NOTE: This is not a guarantee that the item will be stored in the preferred environment.
+                In cases where storage space is limited, the item may be stored in a different environment.""",
+        examples = ["NONE", "FRYS"]
+    )
+    val preferredEnvironment: Environment,
+    @Schema(
+        description = """Whether the item is a single object or a container with other items inside.
+                "NONE" is for single objects, "ABOX" is for archival boxes, etc.
+                NOTE: It is up to the catalogue to keep track of the items inside a container.""",
+        examples = ["NONE", "BOX", "ABOX", "CRATE"]
+    )
+    val packaging: Packaging,
+    @Schema(
+        description = """Callback URL to use for sending item updates to the host system.
+            For example when item moves or changes quantity in storage.""",
+        example = "https://callback-wls.no/item"
+    )
+    val callbackUrl: String?
+) {
+    fun toItem(): Item =
+        Item(
+            hostId = hostId,
+            hostName = hostName,
+            description = description,
+            itemCategory = itemCategory,
+            preferredEnvironment = preferredEnvironment,
+            packaging = packaging,
+            callbackUrl = callbackUrl,
+            location = null,
+            quantity = null
```

Comment:
Quantity can be set to 0, as it helps with checks and not having to worry about NPE

Reply:
Yeah it goes back to the other comment I made about getting rid of nulls and just setting default values. If we don't do it, then we still have to deal with nulls